### ISSUE TITLE
boringssl FallbackSCSV-VersionMatch failure

### ISSUE THREAD

**Issue by kroeckx** [openssl/openssl#2099]
Travis is failing the boringssl test suite with:

```
go test: FAILED (FallbackSCSV-VersionMatch)
go test: unexpected failure: local error 'remote error: alert(86)', child error 'exit status 1', stdout:
go test: 
go test: stderr:
go test: Connection 1 failed.
go test: 47029203250688:error:140A1175:SSL routines:ssl_bytes_to_cipher_list:inappropriate fallback:ssl/statem/statem_srvr.c:3591:
```

This is probably related to enabling TLS 1.3 in the coverage target.

**Comment by levitte** [comment]
@davidben, do you have any idea on the matter?

**Comment by davidben** [comment]
If you see FALLBACK_SCSV, it means "abort if you didn't negotiate your highest version". The test is asserting that if you see a TLS 1.3 ClientHello (and thus negotiate TLS 1.3) with FALLBACK_SCSV, you do not abort. (That should probably get renamed to FallbackSCSV-VersionMatch-TLS13 to align with FallbackSCSV-VersionMatch-TLS12.)

Is it possible your FALLBACK_SCSV code gets confused by the new version negotiation?

**Comment by kroeckx** [comment]
I had a look at the code, and it's at least not obvious what goes wrong.

**Comment by davidben** [comment]
Oh. The problem is you are implementing draft 18 and the snapshot of BoringSSL you're using was back when we were draft 16. We're draft 18 now, so you'll want to roll the BoringSSL revision forward and then we'll match. Though if I do that I get a different failure. Are you all still at a hybrid 1.2/1.3 stack or are you at the point where you can interoperate now?

(FALLBACK_SCSV doesn't quite work right with draft versions due to what's effectively a discontinuity in the version range. There's probably something to be said for defining it to ignore draft versions. Though every indication so far is that the fallback will not be needed now that we've changed version negotiation.)

**Comment by levitte** [comment]
Ok!  What revision do you recommend we pick?

**Comment by levitte** [comment]
When I try "at home", I get a slew of these:

```
go test: profiling:/home/levitte/gitwrk/openssl.net/official/_build/crypto/dh/dh_asn1.gcda:Version mismatch - expected 504* got 602*
```

I have no clue how to deal with that...

**Comment by richsalz** [comment]
it's still very much WIP and interops with nobody.

**Comment by davidben** [comment]
@levitte: Any reasonably recent revision will have draft 18. I'd probably just suggest pulling to latest master whenever you feel like updating. The pinning is just so things don't change out from underneath you.

That said, per @richsalz' comment, if the TLS 1.3 implementation is still WIP, pulling it forward won't help until it's ready to interop. (The test is misnamed and should be called FallbackSCSV-VersionMatch-TLS13 to be a little clearer. I'll go rename it...)

**Comment by levitte** [comment]
Ok, getting past the profiling error and trying with the latest from boring, I get a slew of failures.  Including the one mentioned here (but with a different message)

**Comment by davidben** [comment]
The failure's different because now we get further along and fail because your TLS 1.3 draft 18 stack is still incomplete.

To that end, I bet the slew of failures is because tests which were previously passing by negotiating TLS 1.2 are now negotiating TLS 1.3 draft 18 and hitting the interop failure.

**Comment by levitte** [comment]
Yes, that seems plausible.  Also, I've cleared a few by adding to ErrorMap in `ossl_config.json`, like you suggested earlier.

**Comment by mattcaswell** [comment]
We should not be enabling tls1.3 in conjunction with the external tests. The TLS1.3 support is just not ready for prime time yet.

**Comment by levitte** [comment]
Oh, right.  I think this was fixed with #2123
